x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsPersisterTests.java

```diff
@@ -285,15 +281,43 @@ public void testPersistDatafeedTimingStats() {
         verifyNoMoreInteractions(client);
     }
 
-    @SuppressWarnings({"unchecked"})
     private Client mockClient(ArgumentCaptor<BulkRequest> captor) {
+        return mockClientWithResponse(captor, new BulkResponse(new BulkItemResponse[0], 0L));
+    }
+
+    @SuppressWarnings("unchecked")
+    private Client mockClientWithResponse(ArgumentCaptor<BulkRequest> captor, BulkResponse... responses) {
         Client client = mock(Client.class);
         ThreadPool threadPool = mock(ThreadPool.class);
         when(client.threadPool()).thenReturn(threadPool);
         when(threadPool.getThreadContext()).thenReturn(new ThreadContext(Settings.EMPTY));
-        ActionFuture<BulkResponse> future = mock(ActionFuture.class);
-        when(future.actionGet()).thenReturn(new BulkResponse(new BulkItemResponse[0], 0L));
-        when(client.bulk(captor.capture())).thenReturn(future);
+        List<ActionFuture<BulkResponse>> futures = new ArrayList<>(responses.length - 1);
+        ActionFuture<BulkResponse> future1 = makeFuture(responses[0]);
+        for (int i = 1; i < responses.length; i++) {
+            futures.add(makeFuture(responses[i]));
+        }
+        when(client.bulk(captor.capture())).thenReturn(future1, futures.toArray(ActionFuture[]::new));
         return client;
     }
+
+    @SuppressWarnings("unchecked")
+    private ActionFuture<BulkResponse> makeFuture(BulkResponse response) {
```

Comment:
Make `static`.

BTW, nice name. Reminds that with our work we are all making future :)